proposals/infra/INF-0004-inproc-outofproc-compiler-library-support.md

```diff
@@ -0,0 +1,138 @@
+<!-- {% raw %} -->
+
+# In-process only or In-process and Out-of-process support for the compiler library
+
+* Proposal: [0004](INF-0004-inproc-outofproc-compiler-library-support.md)
+* Author(s): [Cooper Partin](https://github.com/coopp)
+* Sponsor: [Cooper Partin](https://github.com/coopp)
+* Status: **Under Consideration**
+* Impacted Project(s): (Clang)
+
+* Issues:
+
+## Introduction
+
+An effort is underway to bring compilation support into clang for HLSL based
+shaders.  A C style library will be built to enable applications to compile
+a shader from their own processes in addition to being able to launching the
+clang process.
+
+There are two proposed ways one could consume the shader compiler library and
+compile a shader today.
+
+* **In Process** - An application links against the shader compiler library to
+bring in support for shader compilation as a static library
+(full implementation) or a dynamic library. Compilation is performed within the
+calling application's process.
+
+The DirectX Shader library does not currently support being linked as a
+static library.  It is a COM interface library that brings in compilation
+support dynamically. Both static and dynamic in-proc solutions will be 
+supported in the HLSL clang based library.
+
+* **Out of Process** - An application links against the shader compiler library and
+uses the apis to compile which create one or more processes on the
+application's behalf to compile one or more shaders. Compilation is performed
+outside the calling application's process and isolated away to a worker
+process.
+
+In-Process must be supported because it is already a method that applications
+use today to consume the DirectX Shader Compiler library.
+
+Out of process is considered an optional feature as the library could ship
+without it in the first release.
+
+This document is a discussion about an out of process design and to evaluate if
+we need to build the system immediately.
```

Comment:
I think that this document needs to pick one of these possibilities:

* in-proc only
* out-of-proc only
* both in- and out-of-proc options provided

Maybe some more detail on what is meant by in- or out-of-proc would help us understand the decision.

FWIW, going in to this, "out-of-proc only" is my preferred option, so this document shouldn't assume that in-proc is required.

Reply:
Picked!